### PR TITLE
Add os_zone module

### PR DESCRIPTION
**Pull request by rcarrillocruz**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

os_zone
##### ANSIBLE VERSION

```
ubuntu@designate:~$ ansible --version
ansible 2.2.0 (devel fb7940fc50) last updated 2016/05/05 17:40:06 (GMT +200)
  lib/ansible/modules/core: (detached HEAD 9db1233521) last updated 2016/05/05 17:40:18 (GMT +200)
  lib/ansible/modules/extras: (add_os_zone 4b27311e24) last updated 2016/05/06 11:02:23 (GMT +200)
  config file = 
  configured module search path = Default w/o overrides
```
##### SUMMARY

This module allows to create OpenStack Designate zone objects

```
ubuntu@designate:~$ cat test_os_zone.yml 

---
- hosts: localhost
  connection: local
  gather_facts: false
  tasks:
  - os_zone:
      cloud: devstack
      state: present
      name: example.net.
      zone_type: primary
      email: test@example.net
      description: Test zone
      ttl: 3600
ubuntu@designate:~$ ansible-playbook -i 'localhost,' test_os_zone.yml -vvv
No config file found; using defaults

PLAYBOOK: test_os_zone.yml *****************************************************
1 plays in test_os_zone.yml

PLAY [localhost] ***************************************************************

TASK [os_zone] *****************************************************************
task path: /home/ubuntu/test_os_zone.yml:6
<localhost> ESTABLISH LOCAL CONNECTION FOR USER: ubuntu
<localhost> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1462532607.14-249869605420089 `" && echo "` echo $HOME/.ansible/tmp/ansible-tmp-1462532607.14-249869605420089 `" )'
<localhost> PUT /tmp/tmpqbiv3_ TO /home/ubuntu/.ansible/tmp/ansible-tmp-1462532607.14-249869605420089/os_zone
<localhost> EXEC /bin/sh -c 'LANG=en_US.UTF-8 LC_ALL=en_US.UTF-8 LC_MESSAGES=en_US.UTF-8 /usr/bin/python /home/ubuntu/.ansible/tmp/ansible-tmp-1462532607.14-249869605420089/os_zone; rm -rf "/home/ubuntu/.ansible/tmp/ansible-tmp-1462532607.14-249869605420089/" > /dev/null 2>&1'
changed: [localhost] => {"changed": true, "invocation": {"module_args": {"api_timeout": null, "auth": null, "auth_type": null, "availability_zone": null, "cacert": null, "cert": null, "cloud": "devstack", "endpoint_type": "public", "key": null, "region_name": null, "timeout": 180, "verify": true, "wait": true}, "module_name": "os_zone"}, "zone": {"action": "CREATE", "attributes": {}, "created_at": "2016-05-06T11:03:28.000000", "description": "Test zone", "email": "test@example.net", "id": "ea8b3cac-58e2-4e88-a799-ed80429d8086", "links": {"self": "http://10.0.3.183:9001/v2/zones/ea8b3cac-58e2-4e88-a799-ed80429d8086"}, "masters": [], "name": "example.net.", "pool_id": "794ccc2c-d751-44fe-b57f-8894c9f5c842", "project_id": "93f1e3cd8b804c098ad3cbee44bf0d9e", "serial": 1462532608, "status": "PENDING", "transferred_at": null, "ttl": 3600, "type": "PRIMARY", "updated_at": null, "version": 1}}

PLAY RECAP *********************************************************************
localhost                  : ok=1    changed=1    unreachable=0    failed=0 
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks again to @rcarrillocruz. This module is going into community review.

We notice that you are one of the maintainers (@emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz) of this module, so when you think it's ready, please comment "shipit" and we will consider it for merging.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Shrews** [comment]
shipit

**Comment by gregdek** [comment]
Thanks again to @rcarrillocruz for this PR, and thanks @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz for reviewing. Marking for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by resmo** [comment]
I am not sure but I don't see any code in this module for check_mode. Is it implemented?

File: cloud/openstack/os_zone.py [hunk 1]
```diff
@@ -0,0 +1,231 @@
...
+
+    try:
+        cloud = shade.openstack_cloud(**module.params)
+        zone = cloud.get_zone(name)
+
+        if module.check_mode:
+            module.exit_json(changed=_system_state_change(state, email,
+                                                          description, ttl,
+                                                          masters, zone))
+
```
Comment (Shrews) [inline cloud/openstack/os_zone.py:183]: Some of these parameters are not defined until further below, so this check will fail.

**Comment by Shrews** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @rcarrillocruz for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by rcarrillocruz** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @rcarrillocruz. To the current maintainers, @emonty, @shrews, @juliakreger, @j2sol, @rcarrillocruz please review according to guidelines (http://docs.ansible.com/ansible/developing_modules.html#module-checklist) and comment with text 'shipit', 'needs_revision' or 'close_me' as appropriate.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by Shrews** [comment]
Let's not pop() params anymore since it affects verbose output.

needs_revision

**Comment by gregdek** [comment]
Thanks @rcarrillocruz for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Pull request merged**